include/oneapi/dpl/pstl/parallel_backend_utils.h

```diff
@@ -301,6 +306,72 @@ __set_symmetric_difference_construct(_ForwardIterator1 __first1, _ForwardIterato
     return __cc_range(__first2, __last2, __result);
 }
 
+template <typename _ValueType, typename _GetNumThreads, typename _GetThreadNum, typename... _Args>
+struct __enumerable_thread_local_storage
+{
+    template <typename... _LocalArgs>
+    __enumerable_thread_local_storage(_LocalArgs&&... __args)
+        : __thread_specific_storage(_GetNumThreads{}()), __num_elements(0), __args(std::forward<_LocalArgs>(__args)...)
+    {
+    }
+
+    // Note: size should not be used concurrently with parallel loops which may instantiate storage objects, as it may
+    // not return an accurate count of instantiated storage objects in lockstep with the number allocated and stored.
+    // This is because the count is not atomic with the allocation and storage of the storage objects.
+    std::size_t
+    size() const
```

Comment:
Does it make sense to declare  this method as `noexcept' ?

Reply:
sure.

Reply:
Is there need to declare it `noexcept` though? Does it impact exception-related expectations of any classes or functions?
Generally, we do not use `noexcept` "just in case" (otherwise it would have been over all the code.

Reply:
I don't know of a direct need.  My guess is that @SergeyKopienko Sergey noticed it on atomic `load()`, and since that is all we do here, to propagate it up.  No strong opinion here from my side.

Reply:
In this case my idea was that we simple use one operation - `std::atomic::load` which declared as `noexcept`
So we can declare this method too.

Reply:
As I said, we can do it in many cases, but usually we do not. Let's not clutter the code with extra details that serve no purpose.

Reply:
ok